lib/ansible/parsing/vault/__init__.py

```diff
@@ -690,32 +704,39 @@ def encrypt(self, plaintext, password):
     def decrypt(self, message, password, version=None):
         message = message.replace(b'\n', '')
 
-        # SPLIT SALT, DIGEST, AND DATA
-        data = unhexlify(message)
-        salt, mac, cryptedData = data.split(b"\n", 2)
-        salt = unhexlify(salt)
-        ciphertext = unhexlify(cryptedData)
-
-        key1, key2, iv = self.gen_key_initctr(password, salt)
+        if version == b'1.1':
+            message = unhexlify(message)
+            salt, mac, ciphertext = message.split(b'\n', 2)
+            salt = unhexlify(salt)
+            mac = unhexlify(mac)
+            ciphertext = unhexlify(ciphertext)
+            key1, key2, iv = self.gen_key_initctr(password, salt, generate_iv=True)
+        else:
+            message = base64.b64decode(message)
+            salt = message[0:32]
+            mac = message[32:64]
+            ciphertext = message[64:]
+            key1, key2, iv = self.gen_key_initctr(password, salt)
 
         # EXIT EARLY IF MAC DOESN'T VALIDATE
         hmac = HMAC.new(key2, ciphertext, SHA256)
-        if not self.is_equal(mac, to_bytes(hmac.hexdigest())):
+        if not self.is_equal(mac, to_bytes(hmac.digest())):
             return None
 
         # SET THE COUNTER AND THE CIPHER
-        ctr = Counter.new(128, initial_value=int(iv, 16))
+        ctr = Counter.new(128, initial_value=iv)
```

Comment:
CTR mode _requires_ no such thing. Its security depends on a given counter value never being reused with the same key, which we ensure by using a new random seed for PBKDF2 every time.